kroxylicious/src/main/java/io/kroxylicious/proxy/bootstrap/FilterChainFactory.java

```diff
@@ -25,25 +24,30 @@
  * New instances are created during initialization of a downstream channel.
  */
 public class FilterChainFactory {
-    private final Configuration config;
+    private final List<FilterDefinition> filterDefinitions;
 
-    public FilterChainFactory(Configuration config) {
-        this.config = Objects.requireNonNull(config);
+    public FilterChainFactory(List<FilterDefinition> filterDefinitions) {
+        validateFilterDefinitions(filterDefinitions);
+        this.filterDefinitions = Optional.ofNullable(filterDefinitions).orElse(List.of());
     }
 
     /**
      * Validate that a collection of FilterDefinitions contains all the required configuration to configure the filter.
      * @param filterDefinitions the candidates to validate.
-     * @return the Set of filter names which fail validation.
+     * @throws InvalidFilterConfigurationException If there are problems.
      */
-    public static Set<String> validateFilterConfiguration(Collection<FilterDefinition> filterDefinitions) {
+    public static void validateFilterDefinitions(Collection<FilterDefinition> filterDefinitions) {
         if (filterDefinitions == null || filterDefinitions.isEmpty()) {
-            return Set.of();
+            return;
         }
-        return filterDefinitions.stream()
+        var invalidFilters = filterDefinitions.stream()
                 .filter(Predicate.not(FilterDefinition::isDefinitionValid))
                 .map(FilterDefinition::type)
                 .collect(Collectors.toSet());
+        if (!invalidFilters.isEmpty()) {
+            throw new InvalidFilterConfigurationException(invalidFilters.stream()
+                    .collect(Collectors.joining(", ", "Invalid config for [", "]")));
```

Comment:
The old message was "missing required config", but we shouldn't assume here that the problem is "missing required" (even if that's the old validation we do right now).